gui-tk/godley.tcl

```diff
@@ -103,10 +103,10 @@ proc openGodley {id} {
         
         
         
-        menu .$id.menubar.options
+        menu .$id.menubar.options  
         .$id.menubar.options add checkbutton -label "Show Values" -variable preferences(godleyDisplay) -command setGodleyDisplay
         .$id.menubar.options add checkbutton -label "DR/CR style" -variable preferences(godleyDisplayStyle) -onvalue DRCR -offvalue sign -command setGodleyDisplay
-        .$id.menubar.options add checkbutton -label "Enable multiple equities" -variable preferences(multipleEquities) -command setGodleyDisplay
+        .$id.menubar.options add checkbutton -label "Enable multiple equity columns" -variable preferences(multipleEquities) -command "toggleEquityColumns $id"
```

Comment:
OK - with this change, you use a different variable rather than one in the preferences array. Or maybe even do away with the TCL variable.

You can use the menu -postcommand option to set the state of the check button variable every time this menu is posted.

Reply:
Thanks. I've done away with the variable and can change the check button's appearance with the -postcommand option in a similar way we do with paste selection:

`
menu .$id.menubar.options -postcommand "toggleEquityColumns $id"
        .$id.menubar.options add checkbutton -label "Show Values" -variable preferences(godleyDisplay) -command setGodleyDisplay
        .$id.menubar.options add checkbutton -label "DR/CR style" -variable preferences(godleyDisplayStyle) -onvalue DRCR -offvalue sign -command setGodleyDisplay
        .$id.menubar.options add checkbutton -label "Enable multiple equity columns" -command "toggleEquityColumns $id"
`

I know I have to use the -textvariable option somewhere in toggleEquityColumns to change the value of $id.multipleEquities too, but haven't succeeded yet, so I have this in the proc now:

`
# sets each individual Godley table mutliple equity column preference
proc toggleEquityColumns id {
    set v $id.multipleEquities
    if {$v==0} {
	 .$id.menubar.edit entryconfigure end -state disabled
	  $id.requestRedraw
    } else {
	 .$id.menubar.edit entryconfigure end -state normal
	  $id.requestRedraw
    }      
}
`

Reply:
Sorry ,the code comment block does not work very well, so I'm repasting the first bit of code:

        menu .$id.menubar.options -postcommand "toggleEquityColumns $id" 
        .$id.menubar.options add checkbutton -label "Show Values" -variable preferences(godleyDisplay) -command setGodleyDisplay
        .$id.menubar.options add checkbutton -label "DR/CR style" -variable preferences(godleyDisplayStyle) -onvalue DRCR -offvalue sign -command setGodleyDisplay
        .$id.menubar.options add checkbutton -label "Enable multiple equity columns" -command "toggleEquityColumns $id"

Reply:
I couldn't find any obvious way of accessing the state of the checkbutton from the menu items interface. Hence, you probably do need the -variable option. But call it something else, not preferences(...).

Reply:
Thanks, I've fixed this but noticed that when there are more than one equity columns only the button widgets on the last column disappears, and not all of them. The number of columns should also automatically reduce to one when this preference is activated, so I'm busy fixing this too...

Reply:
Extra equity columns are now deleted when the preference is reverted from multiple to single equity column mode. I've taken a little longer to commit the code because I ran the test suite to make sure that all tests pass before the commit. I think everything works correctly now.